BEPs/BEP-299.md

```diff
@@ -0,0 +1,94 @@
+
+# 5. Specification
+
+## 5.1 Asset Back to Wallet
+
+Digital assets on the BNB Beacon Chain may not always be stored in a user's wallet in specific situations. For example, when users participate in staking, their BNB is actually transferred to the system account of the staking module until they redeem it. Modules that can cause similar asset transfers include:
+
+   - Staking module.
+   - Timelock module.
+   - Atomic swap module.
+   - Governance module.
+
+The mentioned modules allow users to contribute funds to a shared pool, making it difficult to distinguish an individual user's share. Hence, we need a solution to ensure users can retrieve their funds into their wallets.
+
+Before BC (Blockchain) comes to a halt, the following logic will be executed:
+1. Disable all transactions that would channel user funds into the liquidity pool.
+2. During the endblock phase, the following transactions are automatically triggered to return user funds to their wallets, like unstake, timer unlock, and atomic swap unlock.
+
+Assets in a user's wallet can be in three states: Free, Frozen, and Locked. When calculating the asset balance, we consider the quantities in all three states. So, even if a token is Locked or Frozen when the Beacon Chain halts, it can still be recovered on the BSC chain.
+
+## 5.2 Asset Recover
+### 5.2.1 Asset Tree
+![overview](./assets/bep-299/asset_tree.png)
+
+The user's asset is abstracted as a node, which includes the *account address, symbol, and amount*. We'll exclude assets from non-EOA accounts, including module system accounts and clearly dead addresses. All these nodes are sorted in lexicographical order based on their keys, and then used to construct a Merkle tree. This Merkle tree will be published in the form of a file on GitHub and Greenfield, allowing anyone to verify the accuracy of their assets based on this file. Community members can challenge the accuracy and integrity of this document, and if any errors are found, the core team will make corrections until there are no objections. 
+
+The Merkle tree enables anyone to verify token ownership on the blockchain using a Merkle proof. It ensures that an account possesses a specific number of tokens represented by the symbol "symbol." 
+
+**Note that only tokens that have been linked to BEP20 tokens on BSC will create a Node. Therefore, assets that have not been mirrored to BSC prior to the BC halting can not be restored.**
+
+### 5.2.2 Asset Recovery App
```

Comment:
bnb136ns6lfw4zs5hg4n85vdthaad7hq5m4gtkgf23

Reply:
I think I’m done coding

Reply:
Im so lost ! I don't understand anything of this transition